mixins/harvester-vm/index.js

```diff
@@ -917,6 +971,107 @@ export default {
       }
     },
 
+    async saveAccessCredentials(vm) {
+      if (!vm?.spec || this.isCreate || this.resource !== HCI.VM) {
+        return true;
+      }
+
+      // save
+      const toSave = [];
+
+      for (const row of this.accessCredentials) {
+        let secretRef = row.secretRef;
+
+        if (!secretRef) {
+          secretRef = await this.$store.dispatch('harvester/create', {
+            metadata: {
+              name:            row.secretName,
+              namespace:       vm.metadata.namespace,
+              labels:          { [HCI_ANNOTATIONS.CLOUD_INIT]: 'harvester' },
+              ownerReferences: this.getOwnerReferencesFromVM(vm)
+            },
+            type: SECRET
+          });
+        }
+
+        if (row.source === ACCESS_CREDENTIALS.RESET_PWD) {
+          secretRef.setData(row.username, row.newPassword);
+        }
+
+        if (row.source === ACCESS_CREDENTIALS.INJECT_SSH) {
+          for (const secretId of row.sshkeys) {
+            const keypair = (this.$store.getters['harvester/all'](HCI.SSH) || []).find(s => s.id === secretId);
+
+            secretRef.setData(`${ keypair.metadata.namespace }-${ keypair.metadata.name }`, keypair.spec.publicKey);
+          }
+        }
+
+        toSave.push(secretRef);
+      }
+
+      try {
+        for (const resource of toSave) {
+          await resource.save();
+        }
+      } catch (e) {
+        new Error(`Function(saveAccessCredentials:save) error ${ e }`);
```

Comment:
We return` return Promise.reject(e)` in catch，  I think it can catch the error， Maybe there's something I'm not understanding correctly。

<img width="1673" alt="image" src="https://user-images.githubusercontent.com/24985926/156912929-f9247fac-e953-487d-b24d-c7a661496df2.png">

Reply:
Sorry, my fault, misread the code(I thought it was `throw`ing the error that was created). Should that error be returned in `Promise.reject` then, otherwise it does nothing?

Reply:
Yes, at the moment we only throw this error, because we create the vm first, and then we create one or more secrets, and if there is an error creating a secret, we do not do anything extra at the moment, such as deleting the wrong vm (because most of the time a secret error is probably caused by a network disconnection), we will deal with this kind of problem in the next release (v1.0.2)

Reply:
Nothing happens with the error created on line 1017 though, the `Promise.reject` returns the originally thrown `e` error. Should it be...
```
} catch (e) {
      return Promise.reject(new Error(`Function(saveAccessCredentials:save) error ${ e }`));
}
```
or
```
} catch (e) {
      return Promise.reject(e);
}
```

Reply:
thanks,  Done,  PTAL.